core/templates/pages/exploration-editor-page/editor-navigation/editor-navigation.component.spec.ts

```diff
@@ -36,15 +38,23 @@ describe('Editor Navigation Component', function() {
   var threadDataService = null;
   var userService = null;
   var windowDimensionsService = null;
-
+  var explorationRightsService = null;
+  var editabilityService = null;
+  var explorationSaveService = null;
+  var changeListService = null;
   var explorationId = 'exp1';
   var userInfo = {
     isLoggedIn: () => true
   };
   var isImprovementsTabEnabledAsyncSpy = null;
 
   beforeEach(angular.mock.module('oppia'));
-
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    let ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
```

Comment:
Oh Yeah sorry forgot to add the comment, Hey @srijanreddy98 can you take a look at this? I tried many different ways, but was unable to get rid of the "Angular injector error", I believe maybe we'll have to find the indirect dependencies since i added add the direct ones and was still getting the injector error.

Reply:
Ok, in that case, you can use it.

Reply:
Great thanks!